File: motel-service/src/main/java/com/kltn/motelservice/service/CriteriaService.java
package com.kltn.motelservice.service;

import com.kltn.motelservice.model.CriteriaDTO;
import org.springframework.data.domain.Page;

public interface CriteriaService {
    Page<CriteriaDTO> getCriteriaByEmail(String email, int page, boolean bool);

    CriteriaDTO createCriteria(CriteriaDTO criteriaDTO, String email);

    CriteriaDTO stopCriteria(Long idCriteria);

    CriteriaDTO startCriteria(Long idCriteria);
}

File: motel-app/src/app/service/criteria.service.ts
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {HttpClient} from '@angular/common/http';
import {CriteriaDTO} from '../model/criteriaDTO';

@Injectable({
  providedIn: 'root'
})
export class CriteriaService {
  private apiUrl = '/api';

  constructor(private http: HttpClient) {
  }

  getCriteria(page: number, all: boolean): Observable<CriteriaDTO[]> {
    return this.http.get<CriteriaDTO[]>(this.apiUrl + '/criterias?page=' + page + '&all=' + all);
  }

  stopCriteria(idCriteria: number): Observable<CriteriaDTO> {
    return this.http.put<CriteriaDTO>(this.apiUrl + '/criteria/stop/' + idCriteria, null);
  }

  startCriteria(idCriteria: number): Observable<CriteriaDTO> {
    return this.http.put<CriteriaDTO>(this.apiUrl + '/criteria/start/' + idCriteria, null);
  }

  createCriteria(criteriaDTO: CriteriaDTO): Observable<CriteriaDTO> {
    return this.http.post<CriteriaDTO>(this.apiUrl + '/criteria', criteriaDTO);
  }
}
